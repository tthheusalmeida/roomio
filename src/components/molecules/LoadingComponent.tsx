import Loading from "../atoms/Loading";

export default function LoadingComponent() {
  return (
    <div className="flex flex-col w-full h-full items-center justify-center">
      <Loading />
    </div>
  );
}
