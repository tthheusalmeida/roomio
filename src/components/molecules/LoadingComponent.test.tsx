import { render, screen } from "@testing-library/react";
import LoadingComponent from "./LoadingComponent";

jest.mock("../atoms/LoadingLogo", () => ({
  __esModule: true,
  default: () => <div data-testid="loading-logo" />,
}));

describe("LoadingComponent", () => {
  it("renders the container and LoadingLogo", () => {
    const { asFragment } = render(<LoadingComponent />);

    const logo = screen.getByTestId("loading-logo");
    expect(logo).toBeInTheDocument();

    const container = logo.parentElement;
    expect(container).toHaveClass(
      "flex",
      "flex-col",
      "w-full",
      "h-full",
      "items-center",
      "justify-center"
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
