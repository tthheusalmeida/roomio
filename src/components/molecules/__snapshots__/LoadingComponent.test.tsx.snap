// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingComponent renders the container and LoadingLogo 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col w-full h-full items-center justify-center"
  >
    <div
      data-testid="loading-logo"
    />
  </div>
</DocumentFragment>
`;
